tests/test_hazmat.py

```diff
@@ -45,6 +45,16 @@ class TestVerifyServiceIdentity:
     Simple integration tests for verify_service_identity.
     """
 
+    def test_no_cert_patterns(self):
+        """
+        Empty cert patterns raise a helpful CertificateError.
+        """
+        with pytest.raises(
+            CertificateError,
+            match="Certificate does not contain any `subjectAltName`s.",
+        ):
+            verify_service_identity([], [], [])
```

Comment:
To make the test easier to read.

```suggestion
            verify_service_identity(cert_patterns=[], obligatory_ids=[], optional_ids=[])
```

Just a very minor API comment. Not a blocker at all.

I was expercting that `optional_ids` would be empty by default, so that you don't alwasy have to pass an empty list, if no optional are required. :)

Reply:
I try to not have default values in internal APIs. They're fine for maintaining backwards-compatibility of public ones, but I find them a readability hinderance.